### PR TITLE
BugFix: Consider conformers for radicals in a BDE calculation

### PR DESCRIPTION
**Pull request by oscarwumit**
Previously ARC did not consider conformers for radicals generated in a BDE calculation. This results in a wrong BDE calculation for ibuprofen (see example in PR). Now ARC will also consider conformers for radicals generated in BDE calculations. With this commit, the correct BDEs for ibuprofen are found.

Example job input file:

> project: ibuprofen-thermo-radical-verify
> 
> specific_job_type: bde
> 
> level_of_theory: apfd/svp//apfd/svp
> model_chemistry: bmk/cbsb7
> use_bac: false
> 
> species:
>   - label: ibuprofen
>     smiles: 'CC(C)CC1=CC=C(C(C)C(=O)O)C=C1'
>     multiplicity: 1
>     xyz: |
>       O      -3.75155600    2.21940100   -0.27888300
>       O      -3.80189800    1.13121800   -2.24143400
>       C       3.28604000    0.39274300   -0.06926300
>       C      -3.36005500   -0.13062700   -0.21099100
>       C       2.44716000   -0.87260200    0.23712300
>       C       3.02093100    1.51785900    0.93840000
>       C       4.77872600    0.04167200   -0.12162800
>       C      -4.04514900   -1.36897600   -0.81029300
>       C      -1.84685900   -0.29896200   -0.09791000
>       C       0.95056300   -0.68160200    0.12969300
>       C       0.14083300   -0.58141300    1.26350200
>       C      -1.23519000   -0.38983900    1.15269000
>       C      -1.04168500   -0.39872100   -1.23858300
>       C       0.33067400   -0.58740200   -1.12276400
>       C      -3.66716700    1.10597800   -1.04359600
>       H       2.98757300    0.75177000   -1.06256600
>       H      -3.74855300    0.05557400    0.79268900
>       H       2.75515200   -1.66463700   -0.45495200
>       H       2.69961200   -1.22574600    1.24369600
>       H       3.59861800    2.41140900    0.68506100
>       H       3.31197300    1.21081800    1.94940300
>       H       1.96519000    1.79689500    0.96384800
>       H       5.12595600   -0.33493700    0.84678100
>       H       5.38168300    0.91968100   -0.37016300
>       H       4.98308800   -0.72857900   -0.87122800
>       H      -3.71627500   -1.53330100   -1.83697900
>       H      -5.13150000   -1.24777200   -0.82085900
>       H      -3.79978100   -2.25304300   -0.21850100
>       H       0.59044600   -0.65502400    2.24856200
>       H      -1.83897100   -0.31359200    2.05131100
>       H      -1.49253900   -0.32144600   -2.22179900
>       H       0.93199300   -0.66793700   -2.02319200
>       H      -3.90065000    2.95989500   -0.88635800
>     bdes:
>     - - 5
>       - 18
>     - - 5
>       - 19

Before this commit, ARC's BDE report

>  BDE report for ibuprofen:
>   Pivots           Atoms        BDE (kJ/mol)
>  (5, 18)           C - H           349.40
>  (5, 19)           C - H           361.00

The BDE for C-H (5, 19) was too high because the conformation of the radical have two hydrogens eclipse from each other.

After this commit, ARC's BDE report

>  BDE report for ibuprofen:
>   Pivots           Atoms        BDE (kJ/mol)
>  (5, 18)           C - H           349.06
>  (5, 19)           C - H           349.10

xyz for C-H (5, 19) before this commit:

>  O                  3.39489500    1.84095000   -0.49492500
>  O                  4.08722500    0.37442700    1.04923400
>  C                 -4.10801400    0.12083200    0.30581600
>  C                  2.65056200   -0.39304700   -0.74617000
>  C                 -2.83601200   -0.10691500    1.07766100
>  C                 -4.17270400    1.51099700   -0.34567100
>  C                 -4.40138700   -0.99476600   -0.70930200
>  C                  3.21605800   -1.80169200   -0.57841200
>  C                  1.21028100   -0.29028200   -0.28515500
>  C                 -1.51001300   -0.15833100    0.58223800
>  C                 -1.15144300    0.00593800   -0.78631700
>  C                  0.17184900   -0.05768200   -1.19782400
>  C                  0.87851300   -0.45047200    1.07315700
>  C                 -0.43721900   -0.38627400    1.49418200
>  C                  3.47025100    0.61395300    0.04133500
>  H                 -4.92354800    0.08723800    1.04982000
>  H                  2.69088100   -0.08821600   -1.80382000
>  H                 -2.94039900   -0.25144000    2.15854400
>  H                 -5.15890800    1.67541900   -0.80898900
>  H                 -3.41208200    1.63556600   -1.13047500
>  H                 -4.00977100    2.30606200    0.39769200
>  H                 -3.65524900   -1.02563800   -1.51698600
>  H                 -5.38900300   -0.84441900   -1.17437400
>  H                 -4.40097200   -1.98293800   -0.22474200
>  H                  3.23513900   -2.08571500    0.48313700
>  H                  4.24745500   -1.86367100   -0.95768200
>  H                  2.59643100   -2.52801600   -1.12380800
>  H                 -1.92667500    0.18689900   -1.53115400
>  H                  0.41080900    0.07597900   -2.25736700
>  H                  1.67647500   -0.61644000    1.80301500
>  H                 -0.67061800   -0.51101500    2.55580200
>  H                  3.88874000    2.43377700    0.09369700

xyz for C-H (5, 19) after this commit:

>  O                  3.44827500    1.88789300   -0.29246400
>  O                  4.06882300   -0.02167400   -1.28387600
>  C                 -3.96539700    0.09072200    0.21576700
>  C                  2.79154800   -0.12188300    0.77410200
>  C                 -2.82114400   -0.47235000   -0.56800000
>  C                 -5.24848300   -0.69916600   -0.04166200
>  C                 -4.16362000    1.58142100   -0.10529800
>  C                  3.38446800   -1.50004700    1.05910900
>  C                  1.31854700   -0.18763500    0.42218800
>  C                 -1.45488700   -0.36613100   -0.21986100
>  C                 -1.00559300    0.26536100    0.97415200
>  C                  0.34530200    0.34981900    1.27714900
>  C                  0.89287400   -0.81247700   -0.76534000
>  C                 -0.45134500   -0.89918500   -1.07962100
>  C                  3.52069200    0.55119800   -0.37544000
>  H                 -3.72650400    0.01075300    1.29303200
>  H                  2.90116900    0.53095400    1.65441900
>  H                 -3.05918600   -0.95092800   -1.52544000
>  H                 -5.12507600   -1.76312000    0.21295200
>  H                 -5.53593000   -0.64059600   -1.10482700
>  H                 -6.08786500   -0.30245000    0.55057700
>  H                 -4.41309800    1.71901700   -1.16981500
>  H                 -3.25078100    2.16106600    0.09958600
>  H                 -4.98530000    2.00909300    0.49231400
>  H                  4.44256300   -1.42449500    1.35269200
>  H                  2.82882700   -1.99322800    1.86965000
>  H                  3.33314000   -2.13481800    0.16338300
>  H                 -1.73540800    0.69349900    1.66514400
>  H                  0.65970600    0.84486200    2.20112100
>  H                  1.64028500   -1.22249400   -1.45109600
>  H                 -0.76081200   -1.38548100   -2.00952100
>  H                  3.87972500    2.24073200   -1.08691500

### CODE REVIEW

**Comment by alongd** [comment]
The test expects to find the species `.initial_xyz` attribute. pass it `.conformers[0]` instead.

**Comment by alongd** [comment]
@oscarwumit, rebase for the tests to pass

File: arc/scheduler.py [hunk 1]
```diff
@@ -774,14 +774,18 @@ def _run_a_job(self, job, label):
...
 
-    def run_conformer_jobs(self):
+    def run_conformer_jobs(self, labels=None):
         """
         Select the most stable conformer for each species using molecular dynamics (force fields) and subsequently
         spawning opt jobs at the conformer level of theory, usually a reasonable yet cheap DFT, e.g., b97d3/6-31+g(d,p).
         The resulting conformer is saved in a string format xyz in the Species initial_xyz attribute.
+
+        Args:
+            labels (list): Species labels to run conformer jobs.
```
Comment (alongd) [inline arc/scheduler.py]: Minor: correct to `Labels of specific species to run conformer jobs for. If None, conformer jobs will be spawned for all species corresponding to labels in self.unique_species_labels.`

File: examples/bde/input.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
+# ARC example BDE job input script
```
Comment (alongd) [inline examples/bde/input.yml]: Please change to:
`ARC example BDE (bond dissociation energy) job input script`
just so the first appearance of "BDE" is clear

File: examples/bde/input.yml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+# Calculate the C-H BDE of ibuprofen for C5-H18 and C5-H19
+# The geometry of ibuprofen was previously optimized and specified below
+# In this exmaple ARC will not generate new conformers for ibuprofen
+
+project: BDE_example_ibuprofen
+
+specific_job_type: bde
+
+level_of_theory: apfd/svp//apfd/svp
```
Comment (alongd) [inline examples/bde/input.yml]: It's also enough to just write `level_of_theory: apfd/svp`, it'll populate both sp and opt/freq

File: examples/bde/input.yml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+model_chemistry: bmk/cbsb7 # a dummy model_chemistry required for Arkane processing
```
Comment (alongd) [inline examples/bde/input.yml]: please add to the comment: `, will not affect the BDE result which is a relative calculation`

File: examples/bde/input.yml [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+use_bac: false
+freq_scale_factor: 0.990
```
Comment (alongd) [inline examples/bde/input.yml]: Would you like to also make a PR for RMG/Arkane with this number? If so, add it here: https://github.com/ReactionMechanismGenerator/RMG-Py/blob/master/arkane/statmech.py#L1057

**Comment by oscarwumit** [comment]
Thanks for the suggestions. I've updated the PR with your recommendations. Please let me know if further changes are needed. @alongd

File: examples/bde/input.yml (hunk 5 above)
Comment (oscarwumit) [reply examples/bde/input.yml]: Yes I will add it to Arkane.

**Pull request merged**